lib/Doctrine/DBAL/Connection.php

```diff
@@ -211,6 +211,15 @@ public function __construct(array $params, Driver $driver, Configuration $config
             unset($this->_params['pdo']);
         }
 
+        if (isset($params["platform"])) {
+            if ( ! $params['platform'] instanceof Platforms\AbstractPlatform) {
+                throw DBALException::invalidPlatformSpecified();
```

Comment:
Should at least pass in the `$params['platform']` and let the exception produce a more detailed message about what was given instead.